scripts/upload_test_results/upload_to_github.py

```diff
@@ -0,0 +1,224 @@
+            if ver_name == commit_name:
+                logging.info('Found existing version %s', ver_name)
+                return ver_name
+
+        # if no current version found in versions.json file after loop
+        # interated all names in file -can't upload results, execution ended
+        logging.critical("ERROR: Can't find %s in versions.json file "
+                         "Script execution ended." % commit_name)
+        sys.exit(1)
+    except Exception:
+        logging.critical("ERROR: Can't download versions.json file "
+                         "Script execution ended.")
+        sys.exit(1)
+
+
+def parse_args():
+    # Read arguments from the command line
+    parser = argparse.ArgumentParser(description="Script to upload Test "
+                                     "Results target.xml files to the "
+                                     "test_results repo on Github")
+    parser.add_argument('-c', "--commit_name",
+                        help="Upload daily test results of that commit",
+                        required=True)
+    parser.add_argument('-j', '--res_dir',
+                        help="Directory with all daily test result files",
+                        required=True)
+    parser.add_argument('-g', '--git_repo_dir',
+                        help="Directory to place git repo test_results",
+                        required=True)
+    parser.add_argument('-t', '--title_msg', help="Title message",
+                        required=True)
+    parser.add_argument('-m', '--commit_msg', help="Commit message",
+                        required=True)
+
+    return parser.parse_args()
+
+
+def git_repo_update(ver_folder_name, repo_dir, upstream_repo, origin_repo,
+                    branch_git_title, commit_name, remote_name, origin_name):
+    # Create if necessary or update test_results git repo on your computer
+    if not os.path.exists(repo_dir):
+        empty_repo = Repo.init(repo_dir)
+        upstream = empty_repo.create_remote(remote_name, upstream_repo)
+
+        logging.info('Creating new repository, please wait ...')
+        assert upstream.exists()
+        assert upstream == empty_repo.remotes.upstream == \
+               empty_repo.remotes[remote_name]
+        upstream.fetch()
+
+        # Setup a local tracking branch of a remote branch
+
+        # create local branch "master" from remote "master"
+        empty_repo.create_head('master', upstream.refs.master)
+        # set local "master" to track remote "master
+        empty_repo.heads.master.set_tracking_branch(upstream.refs.master)
+        # checkout local "master" to working tree
+        empty_repo.heads.master.checkout()
+
+        empty_repo.create_remote(origin_name, origin_repo)
+        origin = Repo(os.path.join(repo_dir))
+        logging.info('Push new repo updates to the origin ...')
+        origin.git.push(origin_name, 'master')
+    else:
+        logging.info("Repository already created, let's update master...")
```

Comment:
what if the given directory is not a git repo? Maybe an error telling this? I think it is better to change the workflow, that first it checks if the directory exists (if not create dir), then if it is git repo and if not, initialize?

Reply:
Yes, we can add it, or add note in a Readme file.